lua/mapvote/server/modules/map_vote.lua

```diff
@@ -0,0 +1,134 @@
+util.AddNetworkString( "MapVote_VoteStarted" )
+util.AddNetworkString( "MapVote_VoteCancelled" )
+util.AddNetworkString( "MapVote_ChangeVote")
+util.AddNetworkString( "MapVote_VoteFinished" )
+util.AddNetworkString( "MapVote_PlayerChangedVote")
+util.AddNetworkString( "RTV_Delay" )
+function MapVote.sendToClient(length, mapsInVote)
+    net.Start( "MapVote_VoteStarted" )
+        net.WriteUInt( #mapsInVote, 32 )
+        for _, map in ipairs(mapsInVote) do
+            net.WriteString( map )
+            net.WriteUInt( MapVote.PlayCounts[map] or 0, 32 )
+        end
+
+        net.WriteUInt( length, 32 )
+    net.Broadcast()
+end
+
+function MapVote.isMapAllowed( m )
+    local conf = MapVote.Config
+    local prefixes = conf.MapPrefixes
+
+    local hookResult = hook.Run( "MapVote_IsMapAllowed", m)
+    if hookResult ~= nil then return hookResult end
+
+    if not MapVote.AllowCurrentMap and m == game.GetMap():lower() .. ".bsp" then return false end -- dont allow current map in vote
+    if MapVote.Config.ExcludedMaps[m] then return false end -- dont allow excluded maps in vote
+
+    if MapVote.Config.IncludedMaps[m] then return true end -- skip prefix check if map is in included maps
+
+    for _, v in pairs( prefixes ) do
+        if string.find( m, "^" .. v ) then
+            return true
+        end
+    end
+    return false
+end
+
+function MapVote.Start( length )
+    length = length or MapVote.Config.TimeLimit or 30
+
+    local maps = file.Find( "maps/*.bsp", "GAME" )
+
+    local mapsInVote = {}
+
+    for _, map in RandomPairs( maps ) do
+        map = map:sub( 1, -5 ) -- strip .bsp
+        if MapVote.isMapAllowed( map ) then
+            table.insert( mapsInVote, map )
+
+            if #mapsInVote >= MapVote.Config.MapLimit then break end
+        end
+    end
+
+    MapVote.State.IsInProgress = true
+    MapVote.State.CurrentMaps = mapsInVote
+    MapVote.State.Votes = {}
+
+    MapVote.sendToClient(length, MapVote.State.CurrentMaps)
```

Comment:
You set the var just above here so it should be safe to use the local
```suggestion
    MapVote.sendToClient( length, mapsInVote )
```